### PR TITLE
Change coredump pattern to writeable directory

### PR DESCRIPTION
**Pull request by kyessenov**

### CODE REVIEW

**Comment by ldemailly** [comment]
I appreciate the quick fix but I think it's wrong - we shouldn't use "/etc/istio/" to put cores

the solution should be something like https://github.com/istio/pilot/issues/1029

**Comment by andraxylia** [comment]
We need core file regardless of whether debug is enabled or not. 

I tend to agree with /etc not being the right place to write cores. What is the reason for changing the directory location from /tmp to /etc?

**Comment by kyessenov** [comment]
/etc/istio/proxy is a tmpfs mounted by k8s. It's a rather painful process to change installation files at the moment. this addresses the immediate need to debug segfaults.

**Comment by ldemailly** [comment]
the reason (explained in #1029) is that the fs is currently hardcoded to readonly

I am not sure I agree that optimized builds should generate core dumps in prod (given that fully optimized builds won't have useful stack traces for instance)

but I do strongly disagree with making core dumps in /etc

if we want a quick hack just change readOnly := true to readOnly := false
there is no data demonstrating that has downside

in other words I don't think you should merge this

**Comment by kyessenov** [comment]
This has nothing to do with raw VM setup. We can mount envoy configs into /tmp if that helps. But we should not be writing files inside the docker container.

**Comment by ldemailly** [comment]
> This has nothing to do with raw VM setup.

?? where does it say raw vm anywhere here or in #1029 ?

**Comment by kyessenov** [comment]
I was implying that the choice of directory is not as important as on a real machine. It's all VFS.

**Comment by kyessenov** [comment]
We can change the location in another PR.

**Comment by andraxylia** [comment]
This has nothing to do with raw VM.

Can we try with a persistent volume? It needs to be available even if the pod/container dies.
